### PR TITLE
First attempt at GH actions for CI

### LINKED ISSUE #102: Set up CI/CD
Initially, CI should be set up to run server-side tests on pushing to GH. Also should set up to automatically deploy to prod when tests pass after a commit to master.

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Use GH actions for CI.

Closes #102 

This uses GH actions to handle CI and CD.  It is configured to run server-side tests whenever a push is made to GH.  When a push is made to master, and the `test-server` job passes, it will push the app to Heroku.

@SchaeStewart to fully configure deployment and get working, you'll need to:

1. Disable the current [automatic github deploys set up in Heroku](https://dashboard.heroku.com/apps/campaign-finance-dashboard/deploy/github)
2. Create a Heroku api key. From command line, assuming you're logged in to Heroku via CLI, do `heroku authorizations: create`. Copy the `Token` value.
3. [Set secret env vars in the settings tab](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository) for this repo on GitHub (I don't have access to do this...).  Specifically, you'll need to set the following vars:
    - HEROKU_API_KEY (the token you created above)
    - HEROKU_EMAIL (the email of the Heroku account used to generate the token)
    - HEROKU_APP_NAME (the unique Heroku app name on Heroku)

I've verified that the CI portion of this workflow works but we'll only be able to confirm CD when config is set up and when we push to master.

Finally, I note in passing that GH actions are *slick*! This is the easiest config/setup for CI/CD i've experienced. My condolences Travis and CircleCI...

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        id: cache
+        uses: actions/cache@v2.1.0
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-${{ hashFiles('**/package-lock.json')}}
+      - name: Install dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: cd server && npm install
+      - name: Run dockerized db
+        run: docker-compose -f docker-compose.yml up -d
```
Comment (SchaeStewart) [inline .github/workflows/build-and-test.yml:29]: `docker-compose -f docker-compose.yml up -d postgres-test` Should allow you to only start the test db

**Comment by SchaeStewart** [comment]
Secrets are configured. Do you want to change the pipeline to only use the `postgres-test`?

**Comment by benjaminEwhite** [comment]
Done @SchaeStewart

**Comment by SchaeStewart** [comment]
Merge whenever you are ready

**Pull request merged**